airflow/serialization/serialized_objects.py

```diff
@@ -80,9 +80,9 @@
 
     HAS_KUBERNETES: bool
     try:
-        from kubernetes.client import models as k8s
+        from kubernetes.client import models as k8s  # noqa: TCH004
 
-        from airflow.providers.cncf.kubernetes.pod_generator import PodGenerator
+        from airflow.providers.cncf.kubernetes.pod_generator import PodGenerator  # noqa: TCH004
```

Comment:
The real dark magic happen here

https://github.com/apache/airflow/blob/64a64abd793c5e7eb49ef68e8724c9346c7536d4/airflow/serialization/serialized_objects.py#L446-L448

That is only affect `k8s` and `PodGenerator` so better exclude only this lines from check rather then entire module